### PR TITLE
Migrate FAT project 'com.ibm.ws.messaging.open_jms20context_fat' from…

### PR DESCRIPTION
**Pull request by tbitonti**
Migration of JMS20 Context FAT from commercial: Folder 'dev/com.ibm.ws.messaging.jms20.JMSContext_fat'  migrated as 'com.ibm.ws.messaging.open_jms20context_fat':

Create FAT project.

Enabled all tests, with the following exceptions:
```
    JMSContextTest_118058
    JMSContextTest_118061, except
        testJMSRedelivered_B_SecOff() -- disabled in commercial liberty
        testJMSRedelivered_TCP_SecOff() -- disabled in commercial liberty
    JMSContextTest_118062
        testGetMessageSelector_Null_B_SecOff() -- disabled in commercial liberty
        testGetMessageSelector_Null_TCP_SecOff() -- disabled in commercial liberty
    JMSContextTest_118067
    JMSContextTest_118070
    JMSContextTest_118075
        testCloseConsumerDepth_TcpIp_SecOff() -- disabled in commercial liberty
        testCloseConsumerSubscription_TcpIp_SecOff() -- disabled in commercial liberty
        testNoLocalTrue_TcpIp_SecOff() -- disabled in commercial liberty

    JMSContextTest_118065_TQ
    JMSContextTest_118066_TQ
    JMSContextTest_118068_TQ
    JMSContextTest_118077_TQ

    LiteBucketSet1Test
    LiteBucketSet2Test
        testTopicNameTOPIC_B_SecOff() -- not working; disabled
        testTopicNameTOPIC_TcpIp_SecOff() -- not working; disabled
    LiteBucketSet3Test
```
This is a FAT only update.  No produce or build code was updated.

At the completion of this FAT project migration, the corresponding FAT project commercial liberty will be removed.

### CODE REVIEW

**Comment by tbitonti** [comment]
#libby

**Comment by tbitonti** [comment]
#libby

**Comment by tbitonti** [comment]
#build

**Comment by tbitonti** [comment]
#libby
#build

File: dev/com.ibm.ws.messaging.open_jms20context_fat/publish/files/unused/185047_Bindings.xml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    </jmsQueueConnectionFactory>
+    
+     <jmsQueueConnectionFactory jndiName="jndi_JMS_BASE_QCF" connectionManagerRef="ConMgr6">
+      <properties.wasJms userName="CF1USER" password="junkpassword" /> 
+												
+    </jmsQueueConnectionFactory>
+    
+      
+    
+    <jmsActivationSpec id="jmsapp/jmsmdb/AckModeMessageDrivenBean">
```
Comment (dazavala) [inline dev/com.ibm.ws.messaging.open_jms20context_fat/publish/files/unused/185047_Bindings.xml:68]: Eventually this server configuration file, and all others that contain jmsActivationSpec elements that declare a javax.jms.Queue or Topic as the destinationType, must be transformed.  Server and client xml files that require transformation for messaging tests are declared in the jakarta-xml-master.properties file in section `#Messaging`.

File: dev/com.ibm.ws.messaging.open_jms20context_fat/fat/src/com/ibm/ws/messaging/JMS20/fat/JMSContextTest_118061.java [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+    // 118061_16 Verify set and get operation on Message header field
+    // JMSRedelivered
+
+    // @Test
+    public void testJMSRedelivered_B_SecOff() throws Exception {
+        boolean testResult = runInServlet("testJMSRedelivered_B_SecOff");
+        assertTrue("Test testJMSRedelivered_B_SecOff failed", testResult);
+    }
+
+    //@Test
```
Comment (dazavala) [inline dev/com.ibm.ws.messaging.open_jms20context_fat/fat/src/com/ibm/ws/messaging/JMS20/fat/JMSContextTest_118061.java:286]: Fyi, If you scan for pattern `// @Test` when enabling specific test methods, you might miss this one.

**Review by dazavala — APPROVED** [review]

**Comment by tbitonti** [comment]
#libby

**Comment by tbitonti** [comment]
#build

File: dev/com.ibm.ws.messaging.open_jms20context_fat/fat/src/com/ibm/ws/messaging/JMS20/fat/JMSContextTest_118061.java (hunk 2 above)
Comment (tbitonti) [reply dev/com.ibm.ws.messaging.open_jms20context_fat/fat/src/com/ibm/ws/messaging/JMS20/fat/JMSContextTest_118061.java:286]: Noted.  I'm correcting that comment to '// @Test' to be consistent with the other disabled test.
I checked the original tests, and the two disabled tests were originally disabled.  I'm adding a comment to note that.

File: dev/com.ibm.ws.messaging.open_jms20context_fat/publish/files/unused/185047_Bindings.xml (hunk 1 above)
Comment (tbitonti) [reply dev/com.ibm.ws.messaging.open_jms20context_fat/publish/files/unused/185047_Bindings.xml:68]: Noted.  Very probably, the MDB application is not used.

File: dev/com.ibm.ws.messaging.open_jms20context_fat/fat/src/com/ibm/ws/messaging/JMS20/fat/FATSuite.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        JMSContextTest_118066_TQ.class, // JMSTemporaryQueue App
+        JMSContextTest_118068_TQ.class, // JMSTemporaryQueue App
+        JMSContextTest_118077_TQ.class, // JMSTemporaryQueue App
+})
+public class FATSuite {
+    // Run only during the Jakarta repeat for now.  When
+    // the tests are removed from WS-CD-Open, the pre-jakarta
+    // repeat can be re-enabled in open-liberty.
+
+    @ClassRule
```
[the remark this answers is not in the document]
Comment (tbitonti) [reply dev/com.ibm.ws.messaging.open_jms20context_fat/fat/src/com/ibm/ws/messaging/JMS20/fat/FATSuite.java:58]: Will do in the final delivery.  I have this enabled for both repeats for my initial personal builds.

**Review by dazavala — APPROVED** [review]
Approved changes to enable JMSContextTest_118075.

**Comment by tbitonti** [comment]
#libby
#build

**Review by dazavala — APPROVED** [review]
Let's notify the JMS team about the testTopicNameTOPIC_XXX_SecOff tests and unused files before we reconvene.  You might formally ask for their review so they can respond formally.  Otherwise, everything look great.  Thanks, Tom.

**Pull request merged**